### PR TITLE
Merged gitRequestStatus into status.client and tested error handling

### PR DESCRIPTION
**Pull request by bobrowadam**
@Shahar - Check it out.

### CODE REVIEW

**Comment by shaharke** [comment]
@bobrowadam you need to use my github account which is @shaharke

File: src/app.js [hunk 1]
```diff
@@ -3,17 +3,19 @@ const LISTEN_PORT = 3222;
...
 const app = express();
 
-
 // Get - uses github.client to retrieve github status and sending result to the client.
 app.get('/', function(req, res) {
 	gitClient().then(function(body) {
 		console.log('Got response from gitStat:\n');
 		body.forEach(function(data){
 			console.log(data);
+			return;
```
Comment (shaharke) [inline src/app.js]: No need

File: src/app.js [hunk 2]
```diff
@@ -3,17 +3,19 @@ const LISTEN_PORT = 3222;
...
 		});
 		res.json(body);
-	}, function(error) {
-		console.error(error);
+		return;
+	}).catch(function(e){
```
Comment (shaharke) [inline src/app.js]: Nice!

File: src/app.js [hunk 3]
```diff
@@ -3,17 +3,19 @@ const LISTEN_PORT = 3222;
...
+		console.error(e);
+		return;
```
Comment (shaharke) [inline src/app.js]: No need

File: src/github.client.js [hunk 4]
```diff
@@ -1,14 +1,36 @@
-const request = require('request');
-const promise = require("bluebird");
-const requestGitStatuses = require('./get.git.status');
-// const request = promise.promisify('request');
+const promise = require('bluebird');
+const request = promise.promisifyAll(require('request'));
```
Comment (shaharke) [inline src/github.client.js]: Good!

File: src/github.client.js [hunk 5]
```diff
@@ -1,14 +1,36 @@
...
 
 module.exports = function gitClient() {
-	return new Promise(function(fulfill, reject) {
-		requestGitStatuses().then(function(body) {
-			fulfill(body);
-		}).catch(function(e) {
-			reject(e);
+	let promiseArray = [];
+	promiseArray[0] = request.getAsync("https://status.github.com/api/messages.json");
```
Comment (shaharke) [inline src/github.client.js]: Just push to the array. Why use indexes?

File: src/github.client.js [hunk 6]
```diff
@@ -1,14 +1,36 @@
...
+	promiseArray[1] = request.getAsync("https://status.github.com/api/last-message.json");
+
+	return promise.all(promiseArray).then(function(dataArray) {
+		return new promise(function(fulfill, reject) {
```
Comment (shaharke) [inline src/github.client.js]: `Promise.all` like all other functions in `Promise` already wrap your code in a promise. So no need to create a new promise. To fulfill the promise you return a value. To reject it you throw an error.

File: src/github.client.js [hunk 7]
```diff
@@ -1,14 +1,36 @@
...
+
+			dataArray.forEach(function(data) {
+				if (data.statusCode >= 300) {
+					console.log('data.statusCode >= 300');
+					reject(new Error(`Status Code: ${data.statusCode}`));
+          return;
+				}
+				parsedData = JSON.parse(data.body);
+				// This is for handling an array object inside dataArray:
+				if ((parsedData).length > 1) {
```
Comment (shaharke) [inline src/github.client.js]: Why the double ()?

File: src/github.client.js [hunk 8]
```diff
@@ -1,14 +1,36 @@
...
+						parsedDataArray.push(data);
+					});
+				} else {
+					parsedDataArray.push(JSON.parse(data.body));
+				}
+			});
+			fulfill(parsedDataArray);
+      return;
 		});
+	}).catch(function(e) {
```
Comment (shaharke) [inline src/github.client.js]: No need

**Comment by shaharke** [comment]
@bobrowadam I've added a bunch of comments. All in all the code is good. I think you're missing on some Promise concepts, but we'll work on that when I return. Feel free to add you questions to this PR.

File: src/app.js (hunk 1 above)
Comment (bobrowadam) [inline src/app.js]: Hmm yeah not sure what was I thinking...

File: src/app.js (hunk 3 above)
Comment (bobrowadam) [inline src/app.js]: The return or the console.error? both?

Comment (shaharke) [inline src/app.js]: the`return`

**Comment by bobrowadam** [comment]
@shaharke I added the test/controller.spec.js.
Its not working though :(. I get this message:
"Cannot promisify an API that has normal methods with '%s'-suffix\u000a\u000a"

I think that it fails when  git.client tries to promisify the stub im "proxywiring" it.

File: test/github.client.getStatus.specs.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+const sinonAsPromised = require("sinon-as-promised")(promise);
+const proxyquire = require('proxyquire');
+
+chai.use(chaiAsPromised);
+const expect = chai.expect();
+const callToMessages = "https://status.github.com/api/messages.json";
+const callToLastMessage = "https://status.github.com/api/last-message.json";
+
+describe("#getStatus", () => {
+  const stub = sinon.stub(request, 'getAsync');
```
Comment (shaharke) [inline test/github.client.getStatus.specs.js]: Don't stun `getAsync` but the original API

File: test/github.client.getStatus.specs.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+const promise = require('bluebird');
+const request = promise.promisifyAll(require('request'));
```
Comment (shaharke) [inline test/github.client.getStatus.specs.js]: You don't need to go promisify here

File: test/github.client.getStatus.specs.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
+const promise = require('bluebird');
```
Comment (shaharke) [inline test/github.client.getStatus.specs.js]: Name Name of file should always correspond to module name not functions

**Pull request merged**